Dockerfile

```
@@ -1,38 +1,26 @@
-# Use an official Python base image from the Docker Hub
 FROM python:3.10-slim
 
-# Install git
-RUN apt-get -y update
-RUN apt-get -y install git chromium-driver
+RUN apt-get update
+RUN apt-get install -y git chromium-driver wget gnupg2 libgtk-3-0 libdbus-glib-1-2 dbus-x11 xvfb ca-certificates
 
-# Install Xvfb and other dependencies for headless browser testing
-RUN apt-get update \
-    && apt-get install -y wget gnupg2 libgtk-3-0 libdbus-glib-1-2 dbus-x11 xvfb ca-certificates
-
-# Install Firefox / Chromium
 RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
     && echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google-chrome.list \
     && apt-get update \
     && apt-get install -y chromium firefox-esr
```

Chromium is already installed by `apt-get install chromium-driver`. This might entirely be replaced by `apt-get install -y firefox-esr`.
```suggestion
RUN apt-get install -y firefox-esr
```